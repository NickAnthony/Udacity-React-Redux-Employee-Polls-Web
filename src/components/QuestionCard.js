import { formatTimestamp } from "../utils/helpers";
import { useNavigate } from "react-router";
import PropTypes from "prop-types";

const QuestionCard = ({ questionId, author, timestamp }) => {
  const navigate = useNavigate();
  const handleClick = (e) => {
    e.preventDefault();
    navigate(`/questions/${questionId}`);
  };
  return (
    <div className="question-card-container">
      <div className="question-card-name">{author}</div>
      <div className="question-card-timestamp">
        {formatTimestamp(timestamp)}
      </div>
      <button className="question-card-show-btn" onClick={handleClick}>
        Show
      </button>
    </div>
  );
};

QuestionCard.propTypes = {
  questionId: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  timestamp: PropTypes.number.isRequired,
};

export default QuestionCard;
