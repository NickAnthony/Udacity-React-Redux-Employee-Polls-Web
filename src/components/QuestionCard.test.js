import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import QuestionCard from "./QuestionCard";

describe("QuestionCard", () => {
  it("question card matches the snapshot for a generic question.", () => {
    var component = render(
      <MemoryRouter>
        <QuestionCard
          questionId={"am8ehyc8byjqgar0jgpub9"}
          author={"sarahedo"}
          timestamp={1488579767190}
        />
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
});
