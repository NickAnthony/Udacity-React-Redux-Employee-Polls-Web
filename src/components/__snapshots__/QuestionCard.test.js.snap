// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionCard question card matches the snapshot for a generic question. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="question-card-container"
      >
        <div
          class="question-card-name"
        >
          sarahedo
        </div>
        <div
          class="question-card-timestamp"
        >
          5:22 PM | 3/3/2017
        </div>
        <button
          class="question-card-show-btn"
        >
          Show
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="question-card-container"
    >
      <div
        class="question-card-name"
      >
        sarahedo
      </div>
      <div
        class="question-card-timestamp"
      >
        5:22 PM | 3/3/2017
      </div>
      <button
        class="question-card-show-btn"
      >
        Show
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
